### PR TITLE
6569 uppersecondary school study progress table Pyramus

### PR DESCRIPTION
**Pull request by saaralepisto**
https://github.com/otavanopisto/muikku/pull/6625

### CODE REVIEW

File: rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java [hunk 1]
```diff
@@ -3535,5 +3543,60 @@ private Status checkStudent(Student student) {
 
     return Status.OK;
   }
+  
+  @Path("/students/{STUDENTID:[0-9]*}/subjectChoices")
+  @GET
+  @RESTPermit(StudentPermissions.LIST_STUDENT_SUBJECT_CHOICES)
```
Comment (aviljakaine) [inline rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java:3549]: Does the student need to be able to load their own choices? If so, this needs to move to an inline check where the ownership is checked separately.

File: rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java [hunk 2]
```diff
@@ -3535,5 +3543,79 @@ private Status checkStudent(Student student) {
...
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    
+    User user = userController.findUserById(studentId);
+    
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+
+    
+    if (!restSecurity.hasPermission(new String[] { StudentPermissions.FIND_STUDENT, UserPermissions.USER_OWNER }, student, Style.OR)) {
```
Comment (aviljakaine) [inline rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java:3560]: This check is already inside `checkStudent` method so not needed again.

File: rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java [hunk 3]
```diff
@@ -3535,5 +3543,79 @@ private Status checkStudent(Student student) {
...
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    StaffMember staffMember = userController.findStaffMemberById(sessionController.getUser().getId());
```
Comment (aviljakaine) [inline rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java:3563]: I would drop this and replace the
`if (staffMember != null){` below with
```
if (loggedUser instanceof StaffMember) {
  StaffMember staffMember = (StaffMember) loggedUser;
  ...
```

**Review by aviljakaine — CHANGES_REQUESTED** [review]
Some cleanup

**Review by aviljakaine — APPROVED** [review]
Looks good to me

**Review by pkukkon — APPROVED** [review]

**Review by pkukkon — APPROVED** [review]
Seems okay but TEACHER to CREATE_COURSESTUDENT permission requires that role to be checked in the Pyramus permission matrix (remind everyone to do this after merging to develop).

**Review by aviljakaine — APPROVED** [review]

**Pull request merged**